src/main/kotlin/CheckersModel.kt

```diff
@@ -0,0 +1,113 @@
+import kotlin.math.abs
+
+class CheckersModel() : BaseModel(8) {
+    private var whoMoves = Color.WHITE
+
+    override fun canMove(turn: BaseTurn): Pair<Boolean, Any?> {
+        if (turn.playerColor != whoMoves) {
+            //ходит не тот, кто должен бы
+            return Pair(false, null)
+        }
+        //TODO("Проверка выхода за границы")
+        val squareFrom = board[turn.from.first][turn.from.second]
+        val squareTo = board[turn.to.first][turn.to.second]
+        if (!squareFrom.color || !squareTo.color) {
+            return Pair(false, null)
+        }
+        if (squareFrom.figure != null) {
+            if (squareTo.figure != null) {
+                //клетка "куда" не пустая
+                return Pair(false, null)
+            }
+            if (squareFrom.figure!!.color != turn.playerColor) {
+                //фигура на клетку "откуда" не совпадает с цветом ходящего
+                return Pair(false, null)
+            }
+            val verticals = turn.to.second - turn.from.second
+            val horizontals = turn.to.first - turn.from.first
+            if ((horizontals == if (turn.playerColor == Color.WHITE) 1 else -1) && (abs(verticals) == 1)) {
```

Comment:
Maybe it's a good idea to store this 1/-1 value as a property of color? In Kotlin, enums are also classes.